libs/designer-ui/src/lib/panel/panelheader/panelheader.tsx

```diff
@@ -1,101 +1,169 @@
+  });
+  const panelExpandTitle = intl.formatMessage({
+    defaultMessage: 'Expand',
+    id: 'oZMhX/',
+    description: 'Text of Tooltip to expand',
+  });
+  const buttonText = isCollapsed ? panelExpandTitle : panelCollapseTitle;
+
+  const className: string = css('collapse-toggle', isRight ? 'right' : 'left', isCollapsed && 'collapsed');
+
+  useEffect(() => {
+    if (isCollapsed || !isOutermostPanel || !nodeId) {
+      return;
+    }
+
+    menuButtonRef.current?.focus();
+  }, [isCollapsed, isOutermostPanel, nodeId]);
+
+  return (
+    <Tooltip relationship="label" positioning={'before'} content={buttonText}>
+      <Button
+        id="msla-panel-header-collapse-nav"
+        ref={menuButtonRef}
+        appearance="subtle"
+        icon={<DismissIcon />}
+        className={className}
+        aria-label={buttonText}
+        onClick={toggleCollapse}
+        data-automation-id="msla-panel-header-collapse-nav"
+      />
+    </Tooltip>
+  );
+};
+
+const OverflowButton = (props: PanelHeaderProps): JSX.Element => {
+  const { headerItems } = props;
+
   const intl = useIntl();
 
-  const menuButtonRef = React.createRef<IButton>();
+  const panelHeaderMenuCommands = intl.formatMessage({
+    defaultMessage: 'More commands',
+    id: '0y5eia',
+    description: 'Label for commands in panel header',
+  });
+
+  return (
+    <Menu>
+      <MenuTrigger>
+        <Tooltip relationship={'label'} positioning={'before'} content={panelHeaderMenuCommands}>
+          <Button
+            appearance="subtle"
+            icon={<OverflowIcon />}
+            aria-label={panelHeaderMenuCommands}
+            data-automation-id="msla-panel-header-more-options"
+          />
+        </Tooltip>
+      </MenuTrigger>
+      <MenuPopover>
+        <MenuList>{headerItems}</MenuList>
+      </MenuPopover>
+    </Menu>
+  );
+};
+
+export const PanelHeader = (props: PanelHeaderProps): JSX.Element => {
+  const {
+    nodeData,
+    isCollapsed,
+    isOutermostPanel,
+    headerLocation,
+    noNodeSelected,
+    panelScope,
+    suppressDefaultNodeSelectFunctionality,
+    readOnlyMode,
+    renameTitleDisabled,
+    horizontalPadding,
+    canResubmit,
+    resubmitOperation,
+    onUnpinAction,
+    commentChange,
+    onRenderWarningMessage,
+    onTitleChange,
+    onTitleBlur,
+  } = props;
+
+  const { comment, displayName: title, iconUri: cardIcon, isError, isLoading, nodeId } = nodeData;
+
+  const intl = useIntl();
 
   const resubmitButtonText = intl.formatMessage({
     defaultMessage: 'Submit from this action',
     id: 'I+85NV',
     description: 'Button label for submitting a workflow to rerun from this action',
   });
-  useEffect(() => {
-    menuButtonRef.current?.focus();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [isCollapsed]);
 
-  const isRight = headerLocation === PanelLocation.Right;
+  const unpinButtonText = intl.formatMessage({
+    defaultMessage: 'Unpin Action',
```

Comment:
`Action` might need to be lowercase here

Reply:
I reused the same string as in designer panel:

![image](https://github.com/user-attachments/assets/2a26fac0-5f4c-4505-9395-d7c715a78d13)

But indeed the header menu uses lowercase :|

![image](https://github.com/user-attachments/assets/09787309-ccc1-4179-b2d6-910efac5266a)

Not sure which we want to use..?

Reply:
Using lowercase here, following up after to align others.